cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/handler/CredentialProviderHttpHandler.java

```diff
@@ -114,18 +116,36 @@ public void listProviders(HttpRequest request, HttpResponder responder) {
    * @throws IOException         If transport errors occur.
    */
   @POST
-  @Path("/credentials/identities/validate")
-  public void validateIdentity(FullHttpRequest request, HttpResponder responder)
+  @Path("/namespaces/{namespace-id}/credentials/identities/validate")
+  public void validateIdentity(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace)
       throws BadRequestException, NotFoundException, IOException {
     CredentialIdentity identity = deserializeRequestContent(request, CredentialIdentity.class);
+    NamespaceMeta namespaceMeta;
     try {
-      credentialProvider.validateIdentity(identity);
+      namespaceMeta = namespaceQueryAdmin.get(new NamespaceId(namespace));
+    } catch (Exception e) {
```

Comment:
In case of non-existing namespace, it will return `404` because `HttpExceptionHandler` will check all the causes and return accordingly: https://github.com/cdapio/cdap/blob/71efaf76a405273ee4cad9a67d3c66853806285d/cdap-common/src/main/java/io/cdap/cdap/common/HttpExceptionHandler.java#L57